src/diff.jl

```diff
@@ -1,4 +1,4 @@
-abstract type Operator <: Function end
+abstract type Operator end
```

Comment:
@YingboMa or @shashi do either of you know why this causes things to start requiring `nameof` in show methods? I think I may move this abstract type to SymbolicUtils so it can be handled there, but I don't quite know where it's looking for whether something is truly a function or not and why that makes a difference.